src/runtime/execution.rs

```diff
@@ -568,6 +570,35 @@ impl ExecutionState {
             self.current_span_entered = Some(unsafe { extend_span_entered_lt(self.current_span.enter()) });
         }
     }
+
+    fn set_tag_internal(&mut self, tag: u64) {
+        self.current_mut().set_tag(tag);
+    }
+
+    pub(crate) fn set_tag(tag: u64) {
+        ExecutionState::with(|s| {
+            s.set_tag_internal(tag);
+        });
+    }
```

Comment:
If somebody calls this in an improper context (i.e., where there is no current task), they'll get a very unhelpful error message. There are some options. One would be to do `try_current().expect("something helpful")` here. Another would be to make `current`, `get`, and friends in `ExecutionState` expect instead of unwrap, now that they are used by public functions.

Reply:
I was not completely sure about that. If it is guaranteed that client code can only run when `current_task` is `Some(_)` then it is fine.